### PR TITLE
Delete student from team

### LINKED ISSUE #102: Delete student from team
As a TA, I can delete student from team to change the members of the team.

Command Format: `/delete_student_from_team id/<student_id> module/<moduleCode> tutorial/<tutorialClass>
name/<team_name>`

### PR DESCRIPTION
**Pull request by whelan-low**
Fixes #102.

Usage: /delete_student_from_team module/MODULE_CODE tutorial/TUTORIAL_CLASS team/TEAM_NAME

Note: changed command name from `/delete_from_team` to `/delete_student_from_team` for consistency reasons

Also refactored some of the common team-based messages into a new class TeamMessages.

TODO:
- [x] Add more test cases
- [x] Fix help formatting

### CODE REVIEW

**Review by mahadhir247 — COMMENTED** [review]
LGTM! Good job on delete command 👍 . I did notice there were quite a bit of message refactoring. Perhaps in the future you could create a separate PR for this so as to not make the PR too big.

File: src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByIdCommand.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        TutorialClass tutorialClass = moduleAndTutorialClass.getTutorialClass();
+        ModuleCode module = moduleAndTutorialClass.getModule();
+
+        TutorialTeam team = model.getTutorialTeam(tutorialClass, tutorialTeam);
+
+        Person personToDelete;
+
+        personToDelete = model.searchPersonByPredicate(predicate);
+        if (personToDelete == null) {
+            throw new CommandException(String.format(PersonMessages.MESSAGE_PERSON_EMAIL_NOT_FOUND, studentId));
```
[the remark this answers is not in the document]
Comment (whelan-low) [reply src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByIdCommand.java:57]: Good catch! Gonna fix that now

File: src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                        newModule, tutorialClass, newTeam);
+        DeleteStudentFromTeamByIndexCommand allocateStudentToTeamByIndexCommand =
+                new DeleteStudentFromTeamByIndexCommand(Index.fromZeroBased(0),
+                        newModule, tutorialClass, newTeam);
+        assertEquals(deleteOtherStudentFromTeamByEmailCommand.toString(),
+                deleteOtherStudentFromTeamByEmailCommand.toString());
+        assertEquals(deleteStudentFromTeamByIdCommand.toString(),
+                deleteStudentFromTeamByIdCommand.toString());
+        assertEquals(allocateStudentToTeamByIndexCommand.toString(),
+                allocateStudentToTeamByIndexCommand.toString());
```
Comment (Zack-Tay) [inline src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:128]: are these testcases meaningful? I think you are just comparing 2 same objects which should always be true anyways.

File: src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+        // null --> returns false
+        assertFalse(deleteStudentFromTeamByEmailCommand.equals(null));
+
+        // allocation of a different person --> returns false
+        assertFalse(deleteStudentFromTeamByEmailCommand.equals(deleteOtherStudentFromTeamByEmailCommand));
+
+
+        DeleteStudentFromTeamByIndexCommand deleteOtherStudentFromTeamByIndexCommand =
+                new DeleteStudentFromTeamByIndexCommand(Index.fromZeroBased(1),
```
Comment (Zack-Tay) [inline src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:205]: I think the use of magic numbers should be reduced.

**Review by Zack-Tay — COMMENTED** [review]
I like that the messages have been refactored into their respective folders to make references more noticeable and easy to find. However, I notice a few nits in your PR, do look through them! Thanks for the effort and hardwork!

File: src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                deleteStudentFromTeamByIdCommand.toString());
+        assertEquals(allocateStudentToTeamByIndexCommand.toString(),
+                allocateStudentToTeamByIndexCommand.toString());
+
+    }
+
+    @Test
+    public void execute_deleteStudentFromClassById_success() {
+        newTeam.addStudent(validPerson);
+        assertEquals(1, newTeam.getStudents().size());
```
[the remark this answers is not in the document]
Comment (whelan-low) [reply src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:135]: Good catch. I will alter these accordingly

File: src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java (hunk 2 above)
Comment (whelan-low) [reply src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:128]: I'm just following the test cases used in AB3. I also felt it wasnt necessary but I'm not sure if I should delete

File: src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByEmailCommand.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        Person personToDelete;
+
+        personToDelete = model.searchPersonByPredicate(predicate);
+        if (personToDelete == null) {
+            throw new CommandException(String.format(PersonMessages.MESSAGE_PERSON_EMAIL_NOT_FOUND, email));
+        }
+        if (team == null) {
+            throw new CommandException(String.format(TeamMessages.MESSAGE_TEAM_DOES_NOT_EXIST, tutorialTeam,
+                    tutorialClass));
+        }
```
[the remark this answers is not in the document]
Comment (whelan-low) [reply src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByEmailCommand.java:61]: Alright, will refactor. Thanks

File: src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByIndexCommand.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+        TutorialTeam team = tutorialClass.getTutorialTeam(tutorialClass, tutorialTeam);
+
+        Person personToDelete;
+
+        try {
+            personToDelete = model.getFilteredPersonList().get(targetIndex.getZeroBased());
+        } catch (IndexOutOfBoundsException e) {
+            throw new CommandException(
+                    String.format(PersonMessages.MESSAGE_PERSON_INDEX_NOT_FOUND, targetIndex.getOneBased()));
```
[the remark this answers is not in the document]
Comment (whelan-low) [reply src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByIndexCommand.java:55]: Ah, I thought the index was with reference to the address book list. Alright, will fix this. Thanks!

File: src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamCommand.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        ModuleCode module = getModule();
+        TutorialClass tutorialClass = getTutorialClass();
+        ModuleCode existingModule = model.findModuleFromList(module);
+        TutorialClass existingTutorialClass = model.findTutorialClassFromList(tutorialClass, existingModule);
+        if (existingModule == null) {
+            throw new CommandException(String.format(ModuleMessages.MESSAGE_MODULE_NOT_FOUND, module));
+        }
+        if (existingTutorialClass == null) {
+            throw new CommandException(
+                    String.format(ModuleMessages.MESSAGE_TUTORIAL_DOES_NOT_BELONG_TO_MODULE, tutorialClass, module));
```
[the remark this answers is not in the document]
Comment (whelan-low) [reply src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamCommand.java:58]: Yup, it was found in addStudentToClass. I will refactor it into moduleTutorialPair

**Comment by whelan-low** [comment]
In my latest commit, I have fixed the magic numbers issue, and repaired some implementation errors regarding index, Also refactored getModuleTutorialPair into ModuleTutorialPair class.

**Comment by whelan-low** [comment]
Sorry for making such a large PR

**Review by Jajared — APPROVED** [review]
LGTM!

**Pull request merged**